quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/tracing/OTelTracingProvider.java

```diff
@@ -45,7 +45,7 @@ public class OTelTracingProvider implements TracingProvider {
 
     public OTelTracingProvider(OpenTelemetry openTelemetry) {
         this.openTelemetry = openTelemetry;
-        this.scopes = new ArrayDeque<>();
+        this.scopes = new ConcurrentLinkedDeque<>();
```

Comment:
How do you know which scopes belong to each request?

Reply:
@brunobat The `OTelTracingProvider` is now request-scoped, so every new request creates a new instance of the provider. It means these scopes are attached only to the current request. The root span (parent) is the one created by Quarkus/Vert.x REST instrumentation.

Reply:
excelent!